core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -223,6 +224,19 @@ buildTestOutput opts tree =
           , foldGroup = runGroup
           }
           opts tree
+  where
+    AnsiTricks{getAnsiTricks} = lookupOption opts
+    -- We must ensure these lines don't wrap, otherwise the wrong
+    -- line will be cleared later or the test tree printing will
+    -- itself wrap.
+    withoutLineWrap :: IO () -> IO ()
+    withoutLineWrap m = do
```

Comment:
Shall we use [`bracket`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Control-Exception.html#v:bracket_)?

Reply:
I don't know really what the best practice is here. Printing is quick so I don't see it being interrupted by an async exception. Printing could result in an exception caused by stdout becoming unavailable, but in that case we can't print the `enableLineWrap` code anyway, so there is no recovery.

It does seem natural to use `bracket` though. It's up to you.

Reply:
Did it for now.